### DISCUSSION TITLE
Is there a way to retrieve events for rate limited clients?

### DISCUSSION THREAD

**Discussion by DonaldDatThanhNguyen** [emqx/emqx#13007] [Q&A]
I am attempting to identify the clientId of clients that have exceeded the rate limit in order to generate a notification for it.

Currently, the only means of identification I have found is through the debug log:
> 2024-04-29T14:31:14.217190-07:00 [debug] clientid: mqttx_a20cb0bd_4, msg: pause_time_due_to_rate_limit, peername: 127.0.0.1:55746, needs: [{32,bytes},{1,messages}], time_in_ms: 200

Is it possible to generate events for the rate limiter using the rules engine, similar to CONNECTED and DISCONNECTED events?
 
Alternatively, can this be achieved by implementing a custom plugin? 

Any feedback or guidance would be greatly appreciated.

**Comment by zmstone** [comment]
Hi @DonaldDatThanhNguyen 
No, this is not supported. Rate limiting pauses do no emit events.
